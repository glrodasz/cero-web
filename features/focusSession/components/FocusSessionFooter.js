import PropTypes from 'prop-types'
import { Spacer, Button } from '@glrodasz/components'

const FocusSessionFooter = ({ onClickEndSession }) => {
  return (
    <>
      <Spacer.Vertical size="lg" />
      <Button onClick={onClickEndSession} type="primary" isDisabled>
        Finalizar tu sesi√≥n
      </Button>
    </>
  )
}

FocusSessionFooter.propTypes = {
  onClickEndSession: PropTypes.func.isRequired,
}

export default FocusSessionFooter
