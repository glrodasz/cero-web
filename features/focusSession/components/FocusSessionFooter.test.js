import FocusSessionFooter from './FocusSessionFooter'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

jest.mock('@glrodasz/components', () => {
  const { dummyRender } = require('../../../utils/tests/dummyRender')
  const originalModule = jest.requireActual('@glrodasz/components')
  return {
    ...originalModule,
    Spacer: { Vertical: dummyRender('Spacer.Vertical') },
  }
})

describe('[ features / focusSession / components / FocusSessionFooter ]', () => {
  describe('when `FocusSessionFooter` is mounted', () => {
    it('should render', () => {
      // Arrange
      const props = {
        onClickEndSession: () => {},
      }

      // Act
      const { asFragment } = render(<FocusSessionFooter {...props} />)

      // Assert
      expect(asFragment()).toMatchSnapshot()
    })
  })

  describe('when then `Button` is clicked', () => {
    it('should call `onClickEndSession`', () => {
      // Arrange
      const onClickEndSessionMock = jest.fn()
      const props = {
        onClickEndSession: onClickEndSessionMock,
      }

      // Act
      render(<FocusSessionFooter {...props} />)
      userEvent.click(screen.getByText('Finalizar tu sesi√≥n'))

      // Assert
      expect(onClickEndSessionMock).toHaveBeenCalled()
    })
  })
})
