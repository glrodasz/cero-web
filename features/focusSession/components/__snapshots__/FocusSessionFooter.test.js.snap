// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ features / focusSession / components / FocusSessionFooter ] when \`FocusSessionFooter\` is mounted should render 1`] = `
<DocumentFragment>
  
[Spacer.Vertical {"size":"lg"} /]
  <button
    class="button type-primary"
  >
    <header
      class="heading color-inverted size-md weight-bold"
    >
      Finalizar tu sesi√≥n
    </header>
  </button>
</DocumentFragment>
`;
